ch4/exercise4-2/printHash.go

```diff
@@ -0,0 +1,32 @@
+package main
+
+import (
+	"bufio"
+	"crypto/sha256"
+	"crypto/sha512"
+	"flag"
+	"fmt"
+	"os"
+)
+
+func main() {
+
+	var s384 = flag.Bool("sha384", false, "generate sha384 hash")
+	var s512 = flag.Bool("sha512", false, "generate sha512 hash")
+	flag.Parse()
+	reader := bufio.NewReader(os.Stdin)
+	fmt.Println("Input: ")
+	input, _ := reader.ReadString('\n')
+
+	if *s384 {
+		fmt.Println("Converted to SHA384")
+		fmt.Println(sha512.Sum384([]byte(input)))
```

Comment:
Use https://golang.org/pkg/encoding/hex/#Dump to output hexidecimal.  This is a pretty standard format for hashes (e.g. git commits), whereas int arrays are unwieldy on the CLI.